ui/raidboss/data/07-dt/raid/r4n.ts

```diff
@@ -1,14 +1,274 @@
+
+// TODO: Map out MapEffect data if needed? Might be useful for prep for savage.
+// TODO: Better triggers for Sidewise Spark. Some sort of phase detection and collect setup is needed
+// as well as identifying the clones based on npc ID or something
+// TODO: Same thing for Bewitching Flight, collector for the loop to combine trigger with clone
+// TODO: Witch Hunt, determine starting point and figure out how to word the dodge
+
+// TODO: Might be able to use these to detect phase push, I didn't look into it very much
+const npcYellData = {
+  // Offsets: 456920,494045,510794
+  '43D4': {
+    'yellId': '43D4',
+    'text': 'M-My body...',
+    'npcIds': ['3301'],
+  },
+  // Offsets: 482233,519355,536125
+  '43D5': {
+    'yellId': '43D5',
+    'text': 'Ugh... How is this possible...?',
+    'npcIds': ['3301'],
+  },
+  // Offsets: 507543,544663,561452,569975,595291
+  '43D7': {
+    'yellId': '43D7',
+    'text': '<pant> <pant>',
+    'npcIds': ['3301'],
+  },
+} as const;
+console.assert(npcYellData);
+
+const headMarkerData = {
+  // Vfx Path: com_share3t
+  stack: '00A1',
+  // Vfx Path: com_share5a1
+  multiHitStack: '013C',
+  // Vfx Path: tag_ae5m_8s_0v
+  spread: '0159',
+  // Vfx Path: tank_laser_5sec_lockon_c0a1
+  tankBusterLine: '01D7',
+} as const;
+console.assert(headMarkerData);
 
 const triggerSet: TriggerSet<Data> = {
   id: 'AacLightHeavyweightM4',
   zoneId: ZoneId.AacLightHeavyweightM4,
   timelineFile: 'r4n.txt',
-  triggers: [],
+  initData: () => ({
+    expectedBlasts: 3,
+    storedBlasts: [],
+  }),
+  triggers: [
+    {
+      id: 'R4N Headmarker Soaring Soulpress Stack',
+      type: 'HeadMarker',
+      netRegex: { id: headMarkerData.stack, capture: true },
+      response: Responses.stackMarkerOn(),
+    },
+    {
+      id: 'R4N Headmarker Wicked Bolt Multi Hit Stack',
+      type: 'HeadMarker',
+      netRegex: { id: headMarkerData.multiHitStack, capture: true },
+      response: Responses.stackMarkerOn(),
+    },
+    {
+      id: 'R4N Headmarker Thunderstorm Spread',
+      type: 'HeadMarker',
+      netRegex: { id: headMarkerData.spread, capture: false },
+      suppressSeconds: 5,
+      response: Responses.spread(),
+    },
+    {
+      id: 'R4N Headmarker Wicked Jolt Tankbuster',
+      type: 'HeadMarker',
+      netRegex: { id: headMarkerData.tankBusterLine, capture: true },
+      response: Responses.tankBuster(),
+    },
+    {
+      id: 'R4N Wrath of Zeus',
+      type: 'StartsUsing',
+      netRegex: { id: '92C7', source: 'Wicked Thunder', capture: false },
+      response: Responses.aoe(),
+    },
+    {
+      id: 'R4N Sidewise Spark Go Left',
+      type: 'StartsUsing',
+      netRegex: { id: ['92BC', '92BE'], source: 'Wicked Thunder', capture: false },
+      response: Responses.goLeft(),
+    },
+    {
+      id: 'R4N Sidewise Spark Go Right',
+      type: 'StartsUsing',
+      netRegex: { id: ['92BD', '92BF'], source: 'Wicked Thunder', capture: false },
+      response: Responses.goRight(),
```

Comment:
I'm going to poke at properly resolving the clones for a bit, but if I can't get it resolved in the next 30 minutes or so I'll just disable these triggers for now, since disabling them only when the clones are up is kind of annoying.

If I can't get it to a good state tonight I'll merge this PR with them disabled and follow up with it tomorrow.